middlewares/validators/progresses.js

```diff
@@ -0,0 +1,99 @@
+const joi = require("joi");
+
+const validateCreateProgressRecords = async (req, res, next) => {
+  const baseSchema = joi
+    .object()
+    .strict()
+    .keys({
+      type: joi.string().trim().valid("user", "task").required().messages({
+        "any.required": "Required field 'type' is missing.",
+        "any.only": "Type field is restricted to either 'user' or 'task'.",
+      }),
+      completed: joi.string().trim().required().messages({
```

Comment:
Are we not allowing multi-line text?

Reply:
No, we do allow multi-line text. The purpose of trimming is to ensure there are no trailing or leading whitespaces